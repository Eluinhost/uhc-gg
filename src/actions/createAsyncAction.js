import { createAction } from 'redux-actions';
import { always } from 'ramda';

const emptyPayload = always({});

export const createAsyncAction = (name, options = {}) => {
  const trigger = createAction(
    name,
    options.triggered || emptyPayload,
    options.triggeredMeta || emptyPayload
  );

  trigger.STARTED = createAction(
    `${name}.STARTED`,
    options.started || emptyPayload,
    options.startedMeta || emptyPayload
  );
  trigger.COMPLETED = createAction(
    `${name}.COMPLETED`,
    options.completed || emptyPayload,
    options.completedMeta || emptyPayload
  );

  return trigger;
};
