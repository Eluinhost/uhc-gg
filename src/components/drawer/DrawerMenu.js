import * as React from 'react';
import {
  Image,
  ScrollView,
  View,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { Link } from 'react-router-native';
import logo from '../../../assets/logo.png';
import { List, ListItem } from 'react-native-elements';
import PropTypes from 'prop-types';
import { AppText } from '../app-text';

export class DrawerMenu extends React.PureComponent {
  static propTypes = {
    closeMenu: PropTypes.func.isRequired,
  };

  static styles = StyleSheet.create({
    menu: {
      paddingTop: 15,
      shadowColor: '#000000',
      shadowOpacity: 0.8,
      shadowRadius: 3,
      backgroundColor: '#263238',
    },
    logoWrapper: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
    },
    logo: {
      margin: 10,
    },
    title: {
      fontSize: 40,
    },
    linksList: {
      marginTop: 0,
    },
    linkTitleStyle: {
      fontFamily: 'ShareTech-Regular',
      color: '#ECEFF1',
    },
    linkContainerStyle: {
      backgroundColor: '#37474F',
    },
  });

  render() {
    return (
      <ScrollView scrollsToTop={false} style={DrawerMenu.styles.menu}>
        <Link
          to="/"
          onPress={this.props.closeMenu}
          component={TouchableOpacity}
        >
          <View style={DrawerMenu.styles.logoWrapper}>
            <Image source={logo} style={DrawerMenu.styles.logo} />
            <AppText style={DrawerMenu.styles.title}>uhc.gg</AppText>
          </View>
        </Link>

        <List containerStyle={DrawerMenu.styles.linksList}>
          <Link
            to="/"
            title="Upcoming Matches"
            titleStyle={DrawerMenu.styles.linkTitleStyle}
            component={ListItem}
            containerStyle={DrawerMenu.styles.linkContainerStyle}
            chevronColor="white"
            onPress={this.props.closeMenu}
          />
        </List>
      </ScrollView>
    );
  }
}
