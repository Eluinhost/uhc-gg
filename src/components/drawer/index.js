export { DrawerWithRedux as Drawer } from './DrawerWithRedux';
