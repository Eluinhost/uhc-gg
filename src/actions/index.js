import { createAction } from 'redux-actions';
import { createAsyncAction } from './createAsyncAction';

export const SET_DRAWER_OPEN = createAction('SET_DRAWER_OPEN', isOpen => ({
  isOpen,
}));

export const UPDATE_UPCOMING_MATCHES = createAsyncAction(
  'UPDATE_UPCOMING_MATCHES',
  {
    completed: matches => ({ matches }),
  }
);
