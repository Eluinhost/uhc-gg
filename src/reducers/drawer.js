import { SET_DRAWER_OPEN } from '../actions';
import { handleActions } from 'redux-actions';

export const drawer = handleActions(
  {
    [SET_DRAWER_OPEN]: (state, action) => ({
      isOpen: action.payload.isOpen,
    }),
  },
  { isOpen: false }
);
