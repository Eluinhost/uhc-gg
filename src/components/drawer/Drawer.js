import * as React from 'react';
import PropTypes from 'prop-types';
import { DrawerMenu } from './DrawerMenu';
import { View } from 'react-native';
import DrawerLib from 'react-native-drawer';
import { App } from '../app';

export class Drawer extends React.PureComponent {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    updateIsOpen: PropTypes.func.isRequired,
  };

  static styles = {
    main: {
      paddingLeft: 3,
    },
    mainOverlay: {
      backgroundColor: 'black',
      opacity: 0,
    },
  };

  static tweenHandler = percent => ({
    mainOverlay: { opacity: percent * 0.8 },
  });

  _onCloseStart = () => this.props.updateIsOpen(false);
  _onOpenStart = () => this.props.updateIsOpen(true);

  _renderContent = () => <DrawerMenu closeMenu={this._onCloseStart} />;

  render() {
    return (
      <DrawerLib
        type="overlay"
        content={this._renderContent()}
        open={this.props.isOpen}
        onClose={this._onCloseStart}
        onOpen={this._onOpenStart}
        captureGestures
        tapToClose
        openDrawerOffset={0.2}
        panCloseMask={0.2}
        panOpenMask={0.05}
        closedDrawerOffset={-3}
        styles={Drawer.styles}
        tweenHandler={Drawer.tweenHandler}
      >
        <View style={App.styles.page}>{this.props.children}</View>
      </DrawerLib>
    );
  }
}
