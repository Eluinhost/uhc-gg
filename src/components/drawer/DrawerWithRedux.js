import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import { SET_DRAWER_OPEN } from '../../actions';
import { Drawer } from './Drawer';

export const mapStateToProps = createSelector(
  state => state.drawer.isOpen,
  isOpen => ({ isOpen })
);

export const mapDispatchToProps = dispatch => ({
  updateIsOpen: isOpen => dispatch(SET_DRAWER_OPEN(isOpen)),
});

export const DrawerWithRedux = connect(mapStateToProps, mapDispatchToProps)(
  Drawer
);
