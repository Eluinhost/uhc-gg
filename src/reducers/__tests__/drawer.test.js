import { drawer } from '../drawer';
import { SET_DRAWER_OPEN } from '../../actions';

describe('drawer reducer', () => {
  it('should set isOpen', () => {
    const pre = {
      isOpen: false,
    };

    const post = drawer(pre, SET_DRAWER_OPEN(true));

    expect(post).toEqual({
      isOpen: true,
    });
    expect(post).toMatchSnapshot();
  });
});
