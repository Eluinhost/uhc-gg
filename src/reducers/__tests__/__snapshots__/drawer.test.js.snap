// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawer reducer should set isOpen 1`] = `
Object {
  "isOpen": true,
}
`;
